File: ts-solutions/src/605-can-place-flowers.ts
export function canPlaceFlowers(flowerbed: number[], n: number): boolean {
	let count = 0
	for (let i = 0; i < flowerbed.length; i++) {
		if (flowerbed[i] !== 0) {
			continue
		}
		const canPlaceLeft = i === 0 || flowerbed[i - 1] === 0
		const canPlaceRight = i === flowerbed.length - 1 || flowerbed[i + 1] === 0
		if (canPlaceLeft && canPlaceRight) {
			flowerbed[i] = 1
			count++
			if (count >= n) {
				return true
			}
		}
	}
	return count >= n
}

File: go-solutions/605_can_place_flowers.go
package leetcode

func CanPlaceFlowers(flowerbed []int, n int) bool {
	for i := 0; i < len(flowerbed); i++ {
		if flowerbed[i] == 1 {
			continue
		}
		canPlaceLeft := i == 0 || flowerbed[i-1] == 0
		canPlaceRight := i == len(flowerbed)-1 || flowerbed[i+1] == 0
		if canPlaceLeft && canPlaceRight {
			i += 1
			n--
			if n == 0 {
				return true
			}
		}
	}
	return n <= 0
}
